packages/apps/src/overlays/Connecting.tsx

```diff
@@ -5,12 +5,25 @@ import React from 'react';
 
 import { useApi } from '@polkadot/react-hooks';
 import { settings } from '@polkadot/ui-settings';
+import { Endpoint } from '@polkadot/ui-settings/types';
 
 import { useTranslation } from '../translate';
 import BaseOverlay from './Base';
 
-const wsUrl = settings.apiUrl;
-const isWs = typeof wsUrl === 'string' && wsUrl.startsWith('ws://');
+function networkOrUrl (apiType: Endpoint): string {
+  if (apiType.type === 'json-rpc') {
+    return apiType.param;
+  }
+
+  if (apiType.type === 'substrate-connect') {
+    return apiType.param;
+  }
+
+  return '';
```

Comment:
Not sure why is should/would get to this point? There are only 2 options?

Reply:
This is just to satisfy TS. If removed then error appears:
```
Function lacks ending return statement and return type does not include 'undefined'.ts(2366)
```
Even adding undefined on the return types (and removing that) then different TS warning appears mentioning:
```
Not all code paths return a value.ts(7030)
```

Reply:
So the issue is that to satisfy the linter, we shouldn't put in code that is "weird". So either -

```js
if (type === 'substrate-connect') {
  return 'something';
}

// this is the only other case, so return
return 'rpc-here-we-go';
```

Reply:
... there is something where we can assert never in the default path and TS will check if all are followed, just cannot get an example off-hand now.

Reply:
Well I rechecked the code and (after we changed to param) I dont think there is any need of the `networkOrUrl` function at all.
We could just:
```
const wsUrl = settings.apiType.param;
```